NewForce-Cohort-8/tabloidfullstack-the-body-slam-batallion
# Ah create tag 14

# Description

As an admin I would like to be able to create a new Tag so that I can give authors the ability to better classify their posts.
Given an admin is on the Tag list page
When they select the Create Tag button
Then they should be directed to a form in which they can enter a new Tag name
Given an admin has entered a Tag name
When they click the Save button
Then a new Tag should be saved to the database
And the admin should be directed to the Tag list page



**Created the following files and methods:**

- AddTag()
- AddTag.js


**Related Ticket(s)**
https://trello.com/c/MsaGIKM8

#14 Create a Tag

**Type of Change**


# Testing Instructions

#Setup (can be skipped if you already have the Tabloid database configured with provided seed data)
- Clone this project’s repository to your machine
  - If you’ve already done this and are currently working on a different branch of the project, be sure to `git add` and `git commit` whatever you’re working on on your branch, before you attempt to switch branches
  - Then `git checkout main` to switch to the project’s main branch
- Navigate to the root directory for this repo (yourParentFolderName/tabloidfullstack-the-body-slam-batallion)
- `git fetch -a` to retrieve the current remote-tracked branches for the project, you should see a branch named `AH-createTag-14` as one of the available branches
- `git checkout AH-createTag-14` to switch to my branch
- If you don't see the most recent version of my files, `git pull origin AH-createTag-14`
- `Start TabloidMVC.sln` to open the project in Visual Studio
- If you haven’t created the Tabloid database, please do so in Visual Studio and use the following SQL scripts to populate it accordingly:
  - Start a new query in your database and copy in this SQL script before running to fill out tables and columns: [First SQL Script for Creating DB](https://github.com/NewForce-Cohort-8/tabloidfullstack-the-body-slam-batallion/blob/main/SQL/01_Tabloid_Create_DB.sql)
  - ctrl + a and delete what is in that query and replace it with this SQL script before running to seed the database: [Second SQL Script for Seeding DB](https://github.com/NewForce-Cohort-8/tabloidfullstack-the-body-slam-batallion/blob/main/SQL/02_Tabloid_Seed_Data.sql)


- You will be presented with a login screen. Enter the email foo@bar.com and click "Login" (no password required)

- You will be presented with the welcome screen that simply says "hello"

- Click the Tag Management link in the topside header

-You should see a table that contains all Tags from your database as well as their corresponding Id. This table should be in alphabetical order by Tag Name

-At the bottom of the list you should see a “Create Tag” button

-Click said button

-You will then be redirected to a form “/tag/add”

-Enter “Test” into the input field

-Click “Save” button

-You should then be redirected to the “/tag” list, which should include your new “Test” tag.

-Run this SQL query:

```
SELECT * FROM Tag;
```
 to confirm that your new tag has been added to the database.


# Checklist:
